server/local/factories.ts

```diff
@@ -0,0 +1,57 @@
+import { JSX } from "../../deps/preact.ts"
+import { renderToString } from "../../deps/preact_render_to_string.ts"
+import { serveFile } from "../../deps/std/http/file_server.ts"
+import { Status } from "../../deps/std/http/http_status.ts"
+import * as U from "../../util/mod.ts"
+import { CodePage, FiveHundredPage, FourOFourPage } from "./pages/mod.ts"
+
+export async function staticFile(req: Request, url: URL): Promise<Response> {
+  const { pathname: path } = url
+  if (acceptsHtml(req)) {
+    return page(await CodePage({ path, src: await Deno.readTextFile(url) }))
+  }
+  return await serveFile(req, path)
+}
+
+export async function code(req: Request, path: string, src: string): Promise<Response> {
+  if (acceptsHtml(req)) {
+    return page(await CodePage({ path, src }))
+  }
+  return new Response(src, {
+    headers: { "Content-Type": "application/typescript" },
+  })
+}
+
+export async function redirect(path: string): Promise<Response> {
+  if (path.startsWith("file://")) {
+    return await fetch(path)
+  }
+  return new Response(null, {
+    status: Status.Found,
+    headers: { Location: path },
+  })
+}
+
+export function fourOFour(req: Request): U.PromiseOr<Response> {
+  if (acceptsHtml(req)) {
+    return page(FourOFourPage({}))
+  }
+  return new Response("404", { status: Status.NotFound })
```

Comment:
Even if it accepts HTML it should still get the status code. And the HTML isn't really adding anything.

```suggestion
  return new Response("404", { status: Status.NotFound })
```